app/Http/Controllers/Api/PesanController.php

```diff
@@ -0,0 +1,148 @@
+
+                return response()->json(['status' => true, 'message' => 'Berhasil mengirim pesan' ]);
+            } catch (Exception $e) {
+                return response()->json(['status' => false, 'message' => 'error Exception' ]);
+            }
+        } else {
+            try {
+                DB::transaction(function () use ($request, $desa) {
+                    $id = Pesan::create([
+                        'das_data_desa_id' => $desa->id,
+                        'judul' => $request->get('judul'),
+                        'jenis' => Pesan::PESAN_MASUK,
+                    ])->id;
+
+                    PesanDetail::create([
+                        'pesan_id' => $id,
+                        'text' => Purify::clean($request->get('pesan')),
+                        'pengirim' => $request->pengirim,
+                        'nama_pengirim' => $request->nama_pengirim
+                    ]);
+                });
+
+                return response()->json(['status' => true, 'message' => 'Berhasil mengirim pesan' ]);
+            } catch (Exception $e) {
+                return response()->json(['status' => false, 'message' => 'error Exception' ]);
+            }
+        }
+
+        return response()->json(['result'=>'unknow method']);
+    }
+
+    public function getpesan(GetPesanRequest $request)
+    {
+        // cek desa
+        $desa = DataDesa::where('desa_id', '=', $request->kode_desa)->first();
+
+        if ($desa == null) {
+            return response()->json(['status' => false, 'message' => 'Desa tidak terdaftar' ]);
+        }
+
+        $pesan = Pesan::whereHas('detailPesan', function ($q) use ($request) {
+            $q->where('id', '>', $request->id);
+        })
+        ->with(['detailPesan' => function ($q) use ($request) {
+            $q->where('id', '>', $request->id);
+        }])
+        ->where('das_data_desa_id', $desa->id)
+        ->orWhere('diarsipkan', 1)
+        ->get();
+
+        return response()->json(['status' => true, 'data'=>$pesan]);
+    }
+
+    public function detail(Request $request)
+    {
+        if ($request->has('id')) {
+            $pesan_id = (int) $request->id;
+            $pesan = Pesan::with(['detailPesan'])
+            ->where('id', '=', $pesan_id)
+            ->first();
+
+            return response()->json(['status' => true, 'data'=>$pesan]);
+        }
+        return response()->json(['status' => true, 'message' => 'Tidak ada Pesan untuk ditampilkan' ]);
+    }
+
+    public function setArsipPesan(Request $request)
+    {
+        $array = json_decode($request->get('array_id'));
+        $pesan = Pesan::whereIn('id', $array)->update([
+            'diarsipkan' => Pesan::MASUK_ARSIP
+        ]);
+
+        if ($pesan > 0) {
+            return back()->with('success', 'Pesan berhasil ditandai!');
+        } else {
+            return back()->withInput()->with('error', 'Pesan gagal diarsipkan!');
+        }
```

Comment:
```suggestion
        if ($pesan > 0) {
            return back()->with('success', 'Pesan berhasil ditandai!');
        }
        return back()->withInput()->with('error', 'Pesan gagal diarsipkan!');
```

Mas @apidong, bisa dibuat bgni